pkg/cloud/client.go

```diff
@@ -100,6 +104,6 @@ func (origC *client) NewClientFromSpec(cfg Config) (Client, error) {
 }
 
 func NewClientFromCSAPIClient(cs *cloudstack.CloudStackClient) Client {
-	c := &client{cs: cs, csAsync: cs}
+	c := &client{cs: cs, csAsync: cs, customMetrics: metrics.NewCustomMetrics()}
```

Comment:
Will it be okay to recreate this `customMetrics:` object every time we reconcile? (performance and consistency wise?)

For `r.CSUser` we already create a new client for every reconcile call. But in that case it's created from an existing client, so we could make it reuse the existing `customMetrics:` object. 

Unfortunately when each reconcile request creates its own client it can't do that. So, we may want to initialize a single `customMetrics` object on the base reconciler and pass it to new clients as needed.

Again, this is only if a single consistent `customMetrics:` object is required or creation of said object is somewhat heavy.

Reply:
See my response to your comment about metrics.go line 35, below.